examples/ticks_and_spines/date_precision_and_epochs.py

```diff
@@ -0,0 +1,156 @@
+"""
+=========================
+Date Precision and Epochs
+=========================
+
+Matplotlib can handle `.datetime` objects and `numpy.datetime64` objects using
+a unit converter that recognizes these dates and converts them to floating
+point numbers.
+
+Before Matplotlib 3.3, the default for this conversion returns a float that was
+days since "0000-12-31T00:00:00".  As of Matplotlib 3.3, the default is
+days from "1970-01-01T00:00:00".  This allows more resolution for modern
+dates.  "2020-01-01" with the old epoch converted to 730120, and a 64-bit
+floating point number has a resolution of 2^{-52}, or approximately
+14 microseconds, so microsecond precision was lost.  With the new default
+epoch "2020-01-01" is 10957.0, so the achievable resolution is 0.21
+microseconds.
+
+"""
+import datetime
+import dateutil
+import numpy as np
+
+import matplotlib
+import matplotlib.pyplot as plt
+import matplotlib.dates as mdates
+
+
+def _reset_epoch_for_tutorial():
+    """
+    Users (and downstream libraries) should not use the private method of
+    resetting the epoch.
+    """
+    mdates._reset_epoch_test_example()
+
+
+#############################################################################
+# Datetime
+# --------
+#
+# Python `.datetime` objects have microsecond resolution, so with the
+# old default matplotlib dates could not round-trip full-resolution datetime
+# objects.
+
+old_epoch = '0000-12-31T00:00:00'
+new_epoch = '1970-01-01T00:00:00'
+
+_reset_epoch_for_tutorial()  # Don't do this.  Just for this tutorial.
+mdates.set_epoch(old_epoch)  # old epoch (pre MPL 3.3)
+
+date1 = datetime.datetime(2000, 1, 1, 0, 10, 0, 12,
+                          tzinfo=dateutil.tz.gettz('UTC'))
```

Comment:
I think you should probably use `dateutil.tz.UTC` or `datetime.timezone.utc` here.

Reply:
Great!  Done....